### PR TITLE
Added QProcess to start local server instead fork

### PR DESCRIPTION
**Pull request by mir3x**

### CODE REVIEW

File: client/connectdlg_common.cpp [hunk 1]
```diff
@@ -16,8 +16,9 @@
 
 #include "fc_prehdrs.h"
 
+#include <QDebug>
```
Comment (lmoureaux) [inline client/connectdlg_common.cpp:19]: Useless

File: client/connectdlg_common.cpp [hunk 2]
```diff
@@ -66,25 +56,53 @@
...
+  if (!m_instance) {
+    m_instance = new serverProcess;
+  }
+  return m_instance;
+}
+/* Removes server process */
+void serverProcess::drop()
+{
+  if (m_instance) {
+    delete m_instance;
```
Comment (lmoureaux) [inline client/connectdlg_common.cpp:101]: Direct `delete` on a `QObject` is dangerous (there may be pending events still addressed at this object), use `deleteLater()`.

File: client/connectdlg_common.cpp [hunk 3]
```diff
@@ -274,7 +222,7 @@ bool client_start_server(void)
     return FALSE;
   }
 
-  storage = freeciv_storage_dir();
+  storage = QString(freeciv_storage_dir());
```
Comment (lmoureaux) [inline client/connectdlg_common.cpp:225]: `QString::fromUtf8` for those users with an accent in their name (all Freeciv strings are UTF-8)

File: client/connectdlg_common.cpp [hunk 4]
```diff
@@ -285,299 +233,67 @@ bool client_start_server(void)
...
-    log_verbose("Tried with commandline: '%s'", cmdline3);
-    log_verbose("Tried with commandline: '%s'", cmdline4);
-    output_window_append(ftc_client, _("Couldn't start the server."));
-    output_window_append(ftc_client,
-                         _("You'll have to start one manually. Sorry..."));
-    return FALSE;
+  for (auto trueServer : qAsConst(program)) {
+    serverProcess::i()->start(trueServer, arguments);
+    trueFcser = trueServer;
+    if (serverProcess::i()->waitForStarted(3000) == true) {
```
Comment (lmoureaux) [inline client/connectdlg_common.cpp:269]: (Question not comment) Does this block the event loop or is it in another thread?

**Review by lmoureaux — COMMENTED** [review]
Nice stats +86 -375 :D :D

File: client/connectdlg_common.cpp (hunk 1 above)
Comment (mir3x) [reply client/connectdlg_common.cpp:19]: No, its used by qInfo()

File: client/connectdlg_common.cpp (hunk 4 above)
Comment (mir3x) [reply client/connectdlg_common.cpp:269]: it blocks

**Pull request merged**